tests/System.Text.Primitives.Tests/Encoding/RandomTests.cs

```diff
@@ -725,5 +725,69 @@ public void ConstructingFromUtf16StringWithEmptyStringDoesNotAllocate()
             Assert.True((new Utf8String("")).ReferenceEquals(Utf8String.Empty));
             Assert.True((new Utf8String(string.Empty)).ReferenceEquals(Utf8String.Empty));
         }
+
+        public static IEnumerable<object[]> TryComputeEncodedBytesShouldMatchEncoding_Strings()
+        {
+            string[] data =
+            {
+                "",
+                "abc",
+                "def",
+                "\uABCD",
+                "\uABC0\uABC1\uABC2",
+                "\uABC0bc",
+                "a\uABC1c",
+                "ab\uABC2",
+                "\uABC0\uABC1\uABC2",
+                Encoding.UTF8.GetString(new byte[] { 0xF0, 0x9F, 0x92, 0xA9})
+            };
+            return data.Select(s => new object[] { s });
+        }
+        [Theory]
+        [MemberData("TryComputeEncodedBytesShouldMatchEncoding_Strings")]
+        public void TryComputeEncodedBytesShouldMatchEncoding_Utf8(string value)
+            => TryTryComputeEncodedBytesShouldMatchEncoding(value, TextEncoder.Utf8, Encoding.UTF8);
+
+        [Theory]
+        [MemberData("TryComputeEncodedBytesShouldMatchEncoding_Strings")]
+        public void TryComputeEncodedBytesShouldMatchEncoding_Utf16(string value)
+            => TryTryComputeEncodedBytesShouldMatchEncoding(value, TextEncoder.Utf16, Encoding.Unicode);
+
+        static unsafe void TryTryComputeEncodedBytesShouldMatchEncoding(string value, TextEncoder encoder, Encoding encoding)
+        {
+            int expectedBytes = encoding.GetByteCount(value);
+
+            // test via string input
+            int actual;
+            Assert.True(encoder.TryComputeEncodedBytes(value, out actual));
```

Comment:
@ahsonkhan added - note that I also had to fix the pre-existing internal implementation, as it could throw rather than return false sometimes